firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1844,63 +1844,120 @@ class BrowserViewController: UIViewController,
         }
     }
 
-    private func creditCardAutofillSetup(_ tab: Tab, didCreateWebView webView: WKWebView) {
+    /// Returns whether the credit card autofill feature flag is enabled from Nimbus
+    func autofillCreditCardNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.creditCardAutofillStatus, checking: .buildOnly)
+    }
+
+    /// Returns whether credit card autofill settings are enabled based on user defaults key KeyAutofillCreditCardStatus
+    func autofillCreditCardSettingsUserDefaultIsEnabled() -> Bool {
         let userDefaults = UserDefaults.standard
         let keyCreditCardAutofill = PrefsKeys.KeyAutofillCreditCardStatus
 
-        let autofillCreditCardStatus = featureFlags.isFeatureEnabled(
-            .creditCardAutofillStatus, checking: .buildOnly)
+        return (userDefaults.object(forKey: keyCreditCardAutofill) as? Bool ?? true)
+    }
+
+    /// Returns whether the address autofill feature flag is enabled from Nimbus
+    func addressAutofillNimbusFeatureFlag() -> Bool {
+        return featureFlags.isFeatureEnabled(.addressAutofill, checking: .buildOnly)
+    }
+
+    /// Returns whether address autofill settings are enabled based on user defaults.
+    func addressAutofillSettingsUserDefaultsIsEnabled() -> Bool {
```

Comment:
this can be a private func. If the user default needs to be accessed anywhere else might be worth it to put the logic around the feature flag/user default status into it's own class and made into static vars/functions?

I know the tech team is working on making this BrowserViewController class smaller with less responsibility, so if there's any way to break out any of the logic in this PR into separate classes, it would be good I think.